crates/router/src/workflows/revenue_recovery.rs

```diff
@@ -200,3 +214,159 @@ pub(crate) async fn get_schedule_time_to_retry_mit_payments(
 
     scheduler_utils::get_time_from_delta(time_delta)
 }
+
+#[cfg(feature = "v2")]
+pub(crate) async fn get_schedule_time_for_smart_retry(
+    state: &SessionState,
+    payment_attempt: &PaymentAttempt,
+    payment_intent: &PaymentIntent,
+    retry_count: i32,
+) -> Option<time::PrimitiveDateTime> {
+    let first_error_message = payment_attempt
+        .error
+        .as_ref()
+        .map_or(String::new(), |error| error.message.clone());
+
+    let billing_state = payment_intent
+        .billing_address
+        .as_ref()
+        .and_then(|addr_enc| addr_enc.get_inner().address.as_ref())
+        .and_then(|details| details.state.as_ref())
+        .map(|state_from_address| state_from_address.peek().clone())
+        .unwrap_or_default();
+
+    // Check if payment_method_data itself is None
+    if payment_attempt.payment_method_data.is_none() {
+        logger::debug!(
+            payment_intent_id = ?payment_intent.get_id(),
+            attempt_id = ?payment_attempt.get_id(),
+            message = "payment_attempt.payment_method_data is None"
+        );
+    }
+
+    let card_network_opt = payment_intent
+        .feature_metadata
+        .as_ref()
+        .and_then(|revenue_recovery_data| {
+            revenue_recovery_data
+                .payment_revenue_recovery_metadata
+                .as_ref()
+        })
+        .and_then(|payment_metadata| {
+            payment_metadata
+                .billing_connector_payment_method_details
+                .as_ref()
+        })
+        .and_then(|details| match details {
+            BillingConnectorPaymentMethodDetails::Card(card_info) => card_info.card_network.clone(),
+            _ => None,
+        })
+        .map(|cn| cn.to_string());
+
+    let card_network_str = match card_network_opt {
+        Some(cn_str) => cn_str,
+        None => {
+            return None;
+        }
+    };
+
+    let card_issuer_str = payment_intent
+        .feature_metadata
+        .as_ref()
+        .and_then(|revenue_recovery_data| {
+            revenue_recovery_data
+                .payment_revenue_recovery_metadata
+                .as_ref()
+        })
+        .and_then(|payment_metadata| {
+            payment_metadata
+                .billing_connector_payment_method_details
+                .as_ref()
+        })
+        .and_then(|details| match details {
+            BillingConnectorPaymentMethodDetails::Card(card_info) => card_info.card_issuer.clone(),
+            _ => None,
+        })
+        // will remove this after we are getting the card issuer from Stripe
+        .unwrap_or("CHASE".to_string());
```

Comment:
Ummm, hardcoding things like this is a bad idea, if we forget to remove them later.

Can we consider returning an error from this function maybe?